File: app/src/main/java/andrefigas/com/github/pokemon/AndroidApplication.java
package andrefigas.com.github.pokemon;

import android.app.Application;

import andrefigas.com.github.pokemon.injection.components.ApplicationComponent;
import andrefigas.com.github.pokemon.injection.components.DaggerApplicationComponent;

public class AndroidApplication  extends Application {

    public ApplicationComponent appComponent = DaggerApplicationComponent.create();

}

File: app/src/main/java/andrefigas/com/github/pokemon/AndroidApplication.kt
package andrefigas.com.github.pokemon

import andrefigas.com.github.pokemon.di.modules
import android.app.Application
import org.koin.android.ext.koin.androidContext
import org.koin.core.context.GlobalContext.startKoin


class AndroidApplication : Application() {

    override fun onCreate() {
        super.onCreate()
        startKoin{
            androidContext(this@AndroidApplication)
            modules(listOf(modules))
        }
    }
}